Sources/Commands/SwiftPackageTool.swift

```diff
@@ -341,6 +344,32 @@ extension SwiftPackageTool {
         }
     }
     
+    struct UpdateTemplate: SwiftCommand {
+        static let configuration = CommandConfiguration(
+            abstract: "Update template from remote")
+        
+        @OptionGroup(_hiddenFromHelp: true)
+        var swiftOptions: SwiftToolOptions
+        
+        @Argument(help: "Template to update")
+        var templateName: String
+        
+        func run(_ swiftTool: SwiftTool) throws {
+            guard let configPath = try swiftTool.getConfigPath() else {
+                throw InternalError("Could not find config path")
+            }
+            
+            guard localFileSystem.exists(configPath.appending(components: "templates", "new-package", templateName)) else {
+                throw InternalError("Could not find \(templateName)")
+            }
+            
+            let provider = GitRepositoryProvider()
```

Comment:
probably need to validate that the template is a git repo (it does not have to be!), and throw an error if its not (Cant update)